aim/agent/aid/universes/aci/aci_universe.py

```diff
@@ -518,9 +518,9 @@ def serve(self, context, tenants):
         except Exception as e:
             LOG.error(traceback.format_exc())
             LOG.error('Failed to serve new tenants %s' % tenants)
-            # Rollback served tenants
-            serving_tenants = serving_tenant_copy
-            raise e
+            # There can be orphaned tenant manager objects, If there is a
+            # failure in serve method. Restart the process
+            utils.perform_harakiri(LOG, "Error in serve. Reset the tenants")
```

Comment:
Just a nit - we probably shouldn't say "Reset the tenants", because that's not what's happening here. It's a complete restart of AIM. We might want to instead use the message from line 520 above (and get rid of that log message).

Reply:
Will do